### PR TITLE
Add support for creating tasks to format script files (#337)

### PR DESCRIPTION
**Pull request by Roggstars**
## 🚀 Description
Following #337, I have added two tasks on the project level: `ktfmtCheckScript` and `ktfmtFormatScript`. These tasks will evaluate the formatting of top-level script (`*.kts`) files, like `build.gradle.kts`.

## 📄 Motivation and Context
Currently, script files are ignored by the plugin. Having a consistent formatting between those and source files would be great.

## 🧪 How Has This Been Tested?
I have added (and adjusted) some integration and unit tests. The old tests actually had some formatting issues in the script files that were used to set up the project in a temp directory. I have separated those changes into separate commits for better legibility during review. I will happily squash them into the primary feature commit, if necessary and desired by you.

I have also published the plugin to my local maven and applied it to
- an Android project
- a JVM project
- a multi-module KMP project (with **limitations**, see below)

**Limitations**
As documented in the [README](https://github.com/cortinico/ktfmt-gradle/blob/2102ca3babb8f3b18782c4ce2e44ada95d489dc9/README.md?plain=1#L14), ktfmt is not supposed to be applied to the top-level `build.gradle.kts`. Therefore, the top-level script files are currently not formatted. Is this a limitation of the plugin itself? Do we want to be able to apply the plugin on the top-level (might be useful now that we actually want to format files that are there).

## 📦 Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by simonhauck** [comment]
Thanks for this amazing PR, i will have definitely a look at it later.

There is still one test failing. Could you have a look @Roggstars ?

**Comment by Roggstars** [comment]
I'll check it out, thought I had run the preMerge task locally...

File: plugin-build/plugin/src/main/java/com/ncorti/ktfmt/gradle/KtfmtPluginUtils.kt [hunk 1]
```diff
@@ -63,6 +63,49 @@ internal object KtfmtPluginUtils {
...
     }
 
+    internal fun createScriptTasks(
+        project: Project,
+        projectDir: File,
+        topLevelFormat: TaskProvider<Task>,
+        topLevelCheck: TaskProvider<Task>,
+    ) {
+        val checkTaskName = "${TASK_NAME_CHECK}Script"
+        val formatTaskName = "${TASK_NAME_FORMAT}Script"
```
Comment (cortinico) [inline plugin-build/plugin/src/main/java/com/ncorti/ktfmt/gradle/KtfmtPluginUtils.kt:73]: nit: I'm wondering if those tasks should be pluralized instead:

```suggestion
        val checkTaskName = "${TASK_NAME_CHECK}Scripts"
        val formatTaskName = "${TASK_NAME_FORMAT}Scripts"
```

Comment (Roggstars) [reply plugin-build/plugin/src/main/java/com/ncorti/ktfmt/gradle/KtfmtPluginUtils.kt]: Agreed and resolved in 4bd1fa6f9ddb4228c390763218eda1ffad842870

**Comment by Roggstars** [comment]
I am still not 100% sure why the Windows machine keeps complaining about that single test. It fails because the build.gradle.kts file the test creates and appends is malformatted. I suspect this might have to do with different line separators on the different operating systems, but I am not 100% sure yet. I will check this out on my Windows machine and push a fix.

**Comment by cortinico** [comment]
> Edit: also, sorry for my force-pushes. Wasn't aware of all the messages this would create on the original issue. Will continue working on a branch in my fork now and only push to main when resolved. :)

No problem at all 👍 I'll mark this as draft and please change it back to ready-for-review when you're done

**Comment by Roggstars** [comment]
@cortinico Addressed the failing tests :) Happy to get and address any review/feedback/comments from your end!

The Gradle Wrapper Validation seems to fail right now. Is that something that can be fixed by just re-running?

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
> @cortinico Addressed the failing tests :) Happy to get and address any review/feedback/comments from your end!
> 
> The Gradle Wrapper Validation seems to fail right now. Is that something that can be fixed by just re-running?

Yeah it was a network failure or so

**Pull request merged**